pkg/sethandler/controller.go

```diff
@@ -113,35 +115,71 @@ func (setHandler *SetHandler) PodChanged(oldPod, newPod v1.Pod) {
 	if !shouldPodBeHandled(newPod) {
 		return
 	}
-	setHandler.adjustContainerSets(newPod)
+	if proceed, containersToBeSet := determineContainersToBeSet(oldPod, newPod); proceed {
+			setHandler.adjustContainerSets(newPod, containersToBeSet)
+	}
 }
 
 func shouldPodBeHandled(pod v1.Pod) bool {
 	setterNodeName := os.Getenv("NODE_NAME")
 	podNodeName := pod.Spec.NodeName
-
 	//Pod is not yet scheduled, or it wasn't scheduled to the Node of this specific CPUSetter instance
 	if setterNodeName == "" || podNodeName == "" || setterNodeName != podNodeName {
 		return false
 	}
-	//If the Pod is not running, its containers haven't been created yet - no cpuset cgroup is present to be adjusted by the CPUSetter
-	if pod.ObjectMeta.Annotations[setterAnnotationKey] != "" {
-		return false
-	}
-	if len(pod.Status.ContainerStatuses) == 0 {
-		return false
-	}
-	for _, containerStatus := range pod.Status.ContainerStatuses {
-		if containerStatus.ContainerID == "" {
-			return false
+	return true
+}
+
+func determineContainersToBeSet(oldPod, newPod v1.Pod)  (bool, []string){
+	workingContainers := []string{}
+	found := false
+	if newPod.ObjectMeta.Annotations[setterAnnotationKey] != "" && oldPod.ObjectMeta.Name != ""  {
+		if len(newPod.Status.ContainerStatuses) == 0 || len(oldPod.Status.ContainerStatuses) == 0 {
```

Comment:
is this check rly needed?
1: do we care about oldStatuses?
2: if not you dont need explicitly len check for new statuses, the range will simply see it is empty and skip its body, resulting in you returning what you aynway want to return here in this branch